File: lecture_007(Strings)/l_007_differencebtwcharacters.cpp
#include<iostream>
#include<string>
using namespace std;

string differencebtwchars(string s) {
    for(int i=0; i<s.length()-1; i++) {
        char r1 = s[i];
        char r2 = s[i+1];
        int diff = r2-r1;

        //s = s + r1;     (correct)
        //string diff = to_string(diff1);
        //s = s + diff; 
        
        s.append(i,r1);
        s.append(i,diff);
    }
    //s = s + s[s.length()-1];    (correct)
    //s.append(s[s.length()-1]);   (error)
    s.append(s.length()-1,s[s.length()-1]);
    return s;
}

int main() {
    string s;
    cin>>s;
    //togglecharacters(s);
    //modify1(s);
    cout<<differencebtwchars(s)<<endl;

    return 0;
}

File: lecture_007(Strings)/l_007_differencebtwcharacters.java
import java.util.Scanner;
public class l_007_differencebtwcharacters {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        String s = scn.nextLine();
        System.out.println(differenceBtwChars(s));
    }
    public static String differenceBtwChars(String s) {
        StringBuilder sb = new StringBuilder();
        for(int i=0; i<s.length()-1; i++) {
            char r1 = s.charAt(i);
            char r2 = s.charAt(i+1);
            int diff = r2-r1;

            sb.append(r1);
            sb.append(diff);
        }
        sb.append(s.charAt(s.length()-1));

        return sb.toString();
    }
}